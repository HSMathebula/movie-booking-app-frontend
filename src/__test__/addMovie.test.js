import React from 'react';
import { BrowserRouter as Routers } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import AddMovie from '../components/movies/addMovie';

const user = {
  user: { id: 1, name: 'test' },
};
describe('testing the rendering of the add movie page', () => {
  it('', () => {
    const { container } = render(
      <Provider store={store}>
        <Routers>
          <AddMovie user={user} />
        </Routers>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
