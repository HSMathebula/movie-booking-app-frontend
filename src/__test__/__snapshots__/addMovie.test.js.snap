// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of the add movie page  1`] = `
<div>
  <div
    class="container pt-5"
  >
    <div
      class="text-center mb-5"
    >
      <h1
        class="fw-bold"
      >
        Add Movie
      </h1>
      <p>
        Add new Movie to Bisko.
      </p>
    </div>
    <div
      class="d-flex aligns-items-center justify-content-center h-100 d-inline-block"
    >
      <div
        class="card align-self-center p-4"
        style="width: 40rem;"
      >
        <form>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="title"
            >
              Title
            </label>
            <input
              class="form-control"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <select
              aria-label="Select movie category"
              class="form-select"
            >
              <option
                value=""
              >
                --select categories--
              </option>
            </select>
          </div>
          <div
            class="mb-3"
          >
            <select
              aria-label="Select genre"
              class="form-select"
            >
              <option
                value=""
              >
                --select genres--
              </option>
            </select>
          </div>
          <div
            class="form-floating mb-3"
          >
            <textarea
              class="form-control"
              placeholder="Description"
            />
            <label
              for="description"
            >
              Description
            </label>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="title"
            >
              Duration
            </label>
            <input
              class="form-control"
              type="number"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class="form-control"
                  type="text"
                  value="10/20/2022, 6:15 PM"
                />
              </div>
            </div>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="picture"
            >
              Picture URL
            </label>
            <input
              class="form-control"
              type="text"
              value=""
            />
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </form>
        <div
          class="align-self-end"
        >
          <a
            href="/"
          >
            Home
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
